easybuild/easyblocks/j/java.py

```diff
@@ -129,12 +129,9 @@ def post_processing_step(self):
                 if elf_interp is None:
                     raise EasyBuildError("Failed to isolate ELF interpreter!")
 
-                module_guesses = self.make_module_req_guess()
-
-                bindirs = [os.path.join(self.installdir, bindir) for bindir in module_guesses['PATH'] if
-                           os.path.exists(os.path.join(self.installdir, bindir))]
-                # Make sure these are unique real paths
-                bindirs = list(set([os.path.realpath(path) for path in bindirs]))
+                # Expand paths in PATH and make sure these are unique real paths
+                bindirs = nub([self.expand_module_search_path(bindir) for bindir in self.module_load_environment.PATH])
```

Comment:
This fails with:
```
  File "/tmp/eb-74ctm4vf/included-easyblocks-l2ztkmqk/easybuild/easyblocks/java.py", line 133, in post_processing_step
    bindirs = nub([self.expand_module_search_path(bindir) for bindir in self.module_load_environment.PATH])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/gent/400/vsc40023/easybuild/eb5/easybuild-framework/easybuild/tools/utilities.py", line 218, in nub
    return list(dict.fromkeys(list_))
                ^^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'list'
```

Since `expand_module_search_path` returns a list, we need to "unpack" it:
```suggestion
                bindirs = nub([x for bindir in self.module_load_environment.PATH
                                 for x in self.expand_module_search_path(bindir)])
```